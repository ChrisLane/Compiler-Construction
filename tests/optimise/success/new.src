// This is to test that we can reduce new variables to a value set directly underneath and that values are properly stored.
main() {
    var x = 1;
    x = 2;
    x = 3;
    var y = $x;
}
