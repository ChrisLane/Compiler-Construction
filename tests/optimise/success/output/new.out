
main (  ) { 
    New ( "x" = Const 2 ) In { 
        Asg ( Identifier "x" := Const 3 ) 
    } 
}

